middlewares/team.middleware.js

```diff
@@ -315,6 +375,34 @@ function parseTeam(req, res, next) {
     return next();
 }
 
+/**
+ * @function parsePatch
+ * @param {body: {id: ObjectId}} req 
+ * @param {*} res 
+ * @param {(err?) => void} next 
+ * @return {void}
+ * @description 
+ *      Delete the req.body.id that was added by the validation of route parameter.
+ *      Move attributes belonging to the team schema to req.body.teamDetails.
+ */
+function parsePatch(req, res, next) {
+    delete req.body.id;
+
+    let teamDetails = {};
+
+    for (const val in req.body) {
```

Comment:
Can you explain the design decision behind using this way of copying values into `teamDetails`, as opposed to manually writing it out?

Reply:
For patch, we're not sure on what exactly will be updated. Manually copying values will result in updates of values to be undefined. Previously, this was circumvented by not having a parsePatch, and the middlewares would update using information from req.body. I think that may have been a mistake, as it makes the middleware harder to combine and use in the future, as there may be scenarios where body contains additional information. The goal of this is so that all information that will be the new updated values will be moved to teamDetails. I think this allows good separation of new values and other potential values.

Reply:
Right. Can. we extract this into a general service method that takes original object and a schema, and returns the attributes that match the schema?

Reply:
Made a new issue for this and assigned to you.